fulp_modules/lisa/velvet_fu.dm

```diff
@@ -0,0 +1,292 @@
+
+/datum/martial_art/velvetfu/reset_streak(mob/living/new_target)
+	. = ..()
+
+/datum/martial_art/velvetfu/teach(mob/living/H, make_temporary=0)
+	if(..())
+		to_chat(H, "<span class='userdanger'>You've mastered the arts of Velvet-Fu!</span>")
+		recedingstance.Grant(H)
+		twistedstance.Grant(H)
+
+/datum/martial_art/velvetfu/on_remove(mob/living/H)
+	to_chat(H, "<span class='userdanger'>You've forgotten the arts of Velvet-Fu...'</span>")
+	recedingstance.Remove(H)
+	twistedstance.Remove(H)
+
+/datum/martial_art/velvetfu/proc/check_streak(mob/living/A, mob/living/D)
+	switch(streak)
+		if("receding_stance")
+			streak = ""
+			receding_stance(A,D)
+			return TRUE
+		if("twisted_stance")
+			streak = ""
+			twisted_stance(A,D)
+			return TRUE
+	if(findtext(streak,FLYING_AXEKICK_COMBO))
+		streak = ""
+		flyingAxekick(A,D)
+		return TRUE
+	if(findtext(streak,GOAT_HEADBUTT_COMBO))
+		streak = ""
+		goatHeadbutt(A,D)
+		return TRUE
+	if(findtext(streak,FULL_THRUST_COMBO))
+		streak = ""
+		fullThrust(A,D)
+		return TRUE
+	if(findtext(streak,MINOR_IRIS_COMBO))
+		streak = ""
+		minorIris(A,D)
+	return FALSE
+
+
+/// Receding Stance
+/datum/action/receding_stance
+	name = "Receding Stance - Regenerates Stamina, takes time to do."
+	icon_icon = 'fulp_modules/lisa/sprites/stances.dmi'
+	button_icon_state = "receding_stance"
+	var/stancing = FALSE
+
+/datum/action/receding_stance/Trigger(mob/living/M, mob/living/user)
+	if(owner.incapacitated())
+		to_chat(owner, "<span class='warning'>You can't do stances while incapacitated...</span>")
+		return
+	if(stancing)
+		to_chat(owner, "<span class='warning'>You are already perfoming a stance.</span>")
+		return
+	var/mob/living/carbon/human/H = owner
+	if(owner.mind.martial_art.streak == "receding_stance")
+		owner.visible_message("<span class='danger'>[owner] stops moving back.</i></b>")
+		owner.mind.martial_art.streak = ""
+	else
+		owner.visible_message("<span class='danger'>[owner] moves back and begins to form a stance.</span>", "<b><i>You backpedal and begin to form your stance.</i></b>")
+		stancing = TRUE
+		addtimer(VARSET_CALLBACK(src, stancing, FALSE), 30, TIMER_UNIQUE)
+		if(do_after(owner, 3 SECONDS))
+			owner.visible_message("<span class='danger'>[owner] focuses on his stance.</span>", "<b><i>You focus on your stance. Stamina...</i></b>")
+			owner.mind.martial_art.streak = "receding_stance"
+			H.adjustStaminaLoss(-40)
+		else
+			owner.visible_message("<span class='danger'>[owner] stops moving back.</i></b>")
+			return
+
+/datum/martial_art/velvetfu/proc/receding_stance(mob/living/carbon/user)
+	return // Empty return because we only want the button trigger
+
+/// Twisted Stance
+/datum/action/twisted_stance
+	name = "Twisted Stance - Regenerates a lot of stamina, deals brute damage."
+	icon_icon = 'fulp_modules/lisa/sprites/stances.dmi'
+	button_icon_state = "twisted_stance"
+
+/datum/action/twisted_stance/Trigger(mob/living/M, mob/living/user)
```

Comment:
fair warning, people are going to only pick up this martial art for this action button here and just use normal weapons instead of unarmed attacks/this martial art's shitty combos (aside from maybe goat headbutt)

Reply:
I say have fun to the 1 janitor that buys it.

Reply:
aight, just wanted to make sure you knew